runners/google-cloud-dataflow-java/src/main/java/org/apache/beam/runners/dataflow/DataflowPipelineJob.java

```diff
@@ -250,7 +262,16 @@ State waitUntilFinish(
       }
 
       if (!hasError) {
+        // Reset the backoff.
         backoff.reset();
```

Comment:
Rather than resetting, would it make sense to just *always* recreate the backoff?